examples/di/InfiAgent-DABench/DABench.py

```diff
@@ -0,0 +1,174 @@
+import json
+from pathlib import Path
+
+from examples.di.requirements_prompt import DABENCH
+from metagpt.const import DABENCH_PATH
+
+
+# This code is referenced from https://github.com/InfiAgent/InfiAgent/blob/main/examples/DA-Agent/eval_closed_form.py
+def evaluate_accuracy_by_question(results):
+    correct = sum("correctness" in result and all(result["correctness"].values()) for result in results)
+    total = len(results)
+    return round(correct / total, 4) if total > 0 else 0
+
+
+def evaluate_accuracy_by_sub_question(results):
+    correct = sum(sum(result["correctness"].values()) for result in results if "correctness" in result)
+    total = sum(len(result["correctness"]) for result in results if "correctness" in result)
+    return round(correct / total, 4) if total > 0 else 0
+
+
+def evaluate_accuracy_proportional_by_sub_question_adjusted(results):
+    total_score = 0
+    for result in results:
+        if "correctness" in result:
+            sub_question_count = len(result["correctness"])
+            score_per_sub_question = 1 / sub_question_count if sub_question_count > 0 else 0
+            question_score = sum(result["correctness"].values()) * score_per_sub_question
+            total_score += question_score
+    return round(total_score / len(results), 4) if results else 0
+
+
+class DABench:
+    def __init__(
+        self,
+        questions_file=Path(DABENCH_PATH) / "da-dev-questions.jsonl",
+        answers_file=Path(DABENCH_PATH) / "da-dev-labels.jsonl",
+        template="",
+    ):
+        # Read questions from a JSONL file
+        with open(questions_file, "r") as file:
+            self.questions = {int(json.loads(line)["id"]): json.loads(line) for line in file}
+
+        # Read answers from a JSONL file
+        with open(answers_file, "r") as file:
+            self.answers = {int(json.loads(line)["id"]): json.loads(line) for line in file}
+
+        self.template = template if template else DABENCH
+
+    def get_question(self, question_id):
+        """Retrieve the question by its id."""
+        return self.questions.get(question_id, "Question not found.")
+
+    def get_prompt(self, question_id):
+        """Retrieve the question by its id."""
+        temp = self.get_question(question_id)
```

Comment:
This name doesn't make much sense. Can you change it to a more meaningful one?

Reply:
done